### ISSUE TITLE
请教长文本分类问题 ,关于Thunews结果.

### ISSUE THREAD

**Issue by hxyshare** [ymcui/Chinese-BERT-wwm#57]
您好,我现在在用BERT做长文本分类的任务,使用的是thunews子数据集,但是结果很差,acc只有0.1左右.
`--train_batch_size=8 
               --max_seq_length=512 
               --learning_rate=2e-5`

使用的就是bert官方tensorflow代码,重写run_classifier.py里面的数据处理类.
其他测评任务也测试了,结果都差不多.但是就是这个长文本的分类任务结果很差,请教一下.

**Comment by ymcui** [comment]
重点查看label处理是否正确。

**Comment by hxyshare** [comment]
> 重点查看label处理是否正确。

感谢回复,我再看下.

**Comment by ymcui** [comment]
@hxyshare 
想起一件事，thucnews数据是按照标签来排列的，所以训练的时候记得要对数据进行随机。
具体位置在main函数里，
```
train_examples = readxxxxxxxx
rng = random.Random(SEED)
rng.shuffle(train_examples)
...
```

**Comment by easywaytodo** [comment]
可以提供下 分类的  dataprocessor么

**Comment by hxyshare** [comment]
@ymcui 
我好像找到原因了,代码里是有shuffle的,确实因为Thucnews的数据排列原因,然后shuffle的时候buffer_size太小.
```
if is_training:
      d = d.repeat()
      d = d.shuffle(buffer_size=100)
```
改为5000或者大一点就好了.
感谢!!

**Comment by easywaytodo** [comment]
改成这样后，acc很低
  def input_fn(params):
    """The actual input function."""
    batch_size = params["batch_size"]

    # For training, we want a lot of parallel reading and shuffling.
    # For eval, we want no shuffling and parallel reading doesn't matter.
    d = tf.data.TFRecordDataset(input_file)
    if is_training:
      d = d.repeat()
      d = d.shuffle(buffer_size=5000)

    d = d.apply(
        tf.contrib.data.map_and_batch(
            lambda record: _decode_record(record, name_to_features),
            batch_size=batch_size,
            drop_remainder=drop_remainder))

    return d
